lib/content-server.ts

```diff
@@ -113,25 +113,49 @@ export class ContentServerStack extends cdk.NestedStack {
             platformVersion: ecs.FargatePlatformVersion.VERSION1_4,
             taskDefinition,
         });
-        service.taskDefinition.executionRole?.grantPassRole(customTaskDefinition.grantPrincipal);
-        service.taskDefinition.taskRole.grantPassRole(customTaskDefinition.grantPrincipal);
 
-        // we will replace this hack when CDK supports EFS in Fargate
-        (service.node.tryFindChild('Service') as ecs.CfnService)?.addPropertyOverride(
-            'TaskDefinition',
-            customTaskDefinition.getResponseField('taskDefinition.taskDefinitionArn'),
-        );
+        // setup cloudwatch agent permissions
+        service.taskDefinition.executionRole!.addManagedPolicy(iam.ManagedPolicy.fromAwsManagedPolicyName('AmazonSSMReadOnlyAccess'));
+        service.taskDefinition.executionRole!.addManagedPolicy(iam.ManagedPolicy.fromAwsManagedPolicyName('CloudWatchAgentServerPolicy'));
+        service.taskDefinition.taskRole.addManagedPolicy(iam.ManagedPolicy.fromAwsManagedPolicyName('CloudWatchAgentServerPolicy'));
 
-        props.listener.addTargets('ContentServer', {
+        const targetGroup = props.listener.addTargets('ContentServer', {
             port: httpPort,
             protocol: elbv2.ApplicationProtocol.HTTP,
-            targets: [ service ],
+            targets: [service],
             healthCheck: {
                 enabled: true,
                 timeout: cdk.Duration.seconds(15),
             },
         });
 
+        // auto scaling
+        const scaling = service.autoScaleTaskCount({
+            minCapacity: 1,
+            maxCapacity: 16
+        });
+        scaling.scaleOnMetric('NetworkScaling', {
+            metric: new cloudwatch.Metric({
+                namespace: 'OpenTuna',
+                metricName: 'net_bytes_sent',
+                dimensions: {
+                    interface: "eth1"
+                }
+            }),
+            scalingSteps: [{
```

Comment:
Let's test the network performance of Fargate in ZHY region, then decide how to use metrics to scaling out/in.

Reply:
why not using percentage for scaling?